kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/DatasetTarget.java

```diff
@@ -85,7 +86,12 @@ public boolean handleExisting(WriteMode writeMode, long lastModForSource,
       }
     }
 
-    boolean exists = !view.isEmpty();
+    boolean ready = false;
+    if(view instanceof ReadiableView) {
+      ready = ((ReadiableView)view).isReady();
+    }
+    // a view exists if it isn't empty, or if it has been marked ready
+    boolean exists = !view.isEmpty() || ready;
```

Comment:
Ready is a cheaper check, so that should go first. Then we can avoid accessing files in some cases.